src/patternfly/components/Table/table-tree-view.scss

```diff
@@ -59,7 +60,14 @@ $pf-c-tree-view--MaxDepth: 10;
   }
 
   > .pf-c-table__check {
-    margin-right: var(--pf-c-table__tree-view-main--c-table__check--MarginRight);
+    align-self: stretch;
+    margin: calc(var(--pf-c-table--cell--PaddingTop) * -1) 0 calc(var(--pf-c-table--cell--PaddingBottom) * -1) 0;
+  }
+
+  > .pf-c-table__check label {
+    height: 100%;
+    padding: var(--pf-c-table--cell--PaddingTop) var(--pf-c-table__tree-view-main--c-table__check--MarginRight) var(--pf-c-table--cell--PaddingBottom) var(--pf-c-table__tree-view-main--c-table__check--PaddingLeft);
```

Comment:
FWIW I ran the change by @mceledonia and we agreed to keep the existing spacing, but add an extra 8px between the expandable toggle and checkbox. This change increases the space between the checkbox and text an additional 8px, too. It may be fine - wdyt @mceledonia @mmenestr 

Additional 8px spacing in this PR

<img width="375" alt="Screen Shot 2022-01-26 at 2 27 33 PM" src="https://user-images.githubusercontent.com/35148959/151242088-b5d21310-bdfb-4c2d-b12b-cb7b84c2caaf.png">

Maintaining the existing space between the checkbox and text

<img width="390" alt="Screen Shot 2022-01-26 at 2 27 49 PM" src="https://user-images.githubusercontent.com/35148959/151242089-b76ac842-6d37-41c6-8fdc-ccf0633fec0d.png">

Reply:
@mceledonia @mmenestr @mcarrano can you review this spacing change?